### ISSUE TITLE
Add support for validator in args

### ISSUE THREAD

**Issue by edefimov** [overblog/GraphQLBundle#174]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | yes
| Version/Branch   | x.y.z

It will be convenient if there is a possibility to add validation rules in _args_ configuration section and _fields_ section for InputObject, like this:
```yaml
updateUserInput:
    type: input-object
    config:
        fields:
            id:
                type: "ID!"
                constraints: 
                    - 
                        class: Symfony\Component\Validator\Constraints\Uuid
                        options:
                            # constraint options
            name:
                type: "String"
                constraints:
                    -
                        class: Symfony\Component\Validator\Constraints\Length
                        options:
                            # constraint options
```

```yaml
User:
    type: object
    config:
        fields:
            accounts:
                type: "[Account]"
                args:
                    regDate:
                        constraints:
                            -
                                class: Symfony\Component\Validator\Constraints\Length
                                options:
                                    # constraint options
```

**Comment by mcg-web** [comment]
Using custom scalar types can also answer to this need ?

**Comment by edefimov** [comment]
Custom scalars can be used, but in case of complex logic user will write a lot of boiler plate code which already released in symfony validation component.

**Comment by akomm** [comment]
I think validation is not something that should be part of resolve. It is different to type validation. Schema defines structure and types. Validation does make sense, but as it is not part of resolve process it should not be part of the schema's config in my opinion. You could serialize your data into a plain DTO (data transfer object) with constraint annotation/yml/xml config for those objects and then just validate them.

**Comment by edefimov** [comment]
I think validation is the part of resolve, since if we expect some kind of data, for example url, and user provided something else, we need to reject resolving of the field. Constraints will simplify validation without writing additional code. As for DTOs - one of the greatest advantages of graphql is the possibility to reduce amount of such objects in software.
The validation topic is also discussed in [GraphQL.js](https://github.com/graphql/graphql-js/issues/361)

**Comment by bhoriuchi** [comment]
@edefimov one minor problem I have run into when using custom scalars for validation is that when you throw a custom error or `GraphQLError` from `parseLiteral` for that matter there is no way to include the argument name that failed validation in the error message since the only argument passed is the `valueAST`

It would be nice for either an `info` object be passed as another argument or `graphql` itself modify the final error message to include information a primitive type like `GraphQLString` would throw if you provided something other than a string. Or something in the middle like a new error type `GraphQLArgumentError` that when thrown adds field name/path data to the error message.

The value here is if you have 2 or more arguments that use the same custom scalar and the request fails validation on more than 1 argument you will get the same generic error message granted with different location data but that gets tedious for the requestor to troubleshoot/debug.

Of course if there is already a way to do this please share.

**Comment by mcg-web** [comment]
I'm closing this because I can't really see the need ether the benefits of implementing this right now. Also this will add useless legacy as it is easy creating customs scalar that use validator. Anyway feel free to reopen if you think that I'm missing something...

**Comment by murtukov** [comment]
@edefimov validation is implemented in 0.13 

https://github.com/overblog/GraphQLBundle/blob/master/docs/validation/index.md
